File: server/src/modules/v1/models/vehicle-category.model.ts
import mongoose from 'mongoose';
import { VehicleCategory } from '../@types/db';
import { createSchema } from '../../../utils';
import { DBCollections } from '../../../utils/constants';

const VehicleCategorySchema = createSchema<VehicleCategory>({
  name: {
    type: String,
  },
  image: {
    type: String,
  },
  map_image: {
    type: String,
  },
  rate_discount_amount: {
    type: Number,
    min: 0,
    default: 0,
  },
});

const vehicleCategoryModel = mongoose.model(
  DBCollections.VehicleCategory,
  VehicleCategorySchema
);

export default vehicleCategoryModel;

File: taxify_driver/lib/data/vehicles/vehicle_category_model.dart
class VehicleCategoryModel {
  final String id;
  final String name;
  final String? image;
  final String? mapImage;
  VehicleCategoryModel({
    required this.id,
    required this.name,
    this.image,
    this.mapImage,
  });

  factory VehicleCategoryModel.fromMap(Map<String, dynamic> map) {
    return VehicleCategoryModel(
      id: map['_id'] as String,
      name: map['name'] as String,
      image: map['image'] != null ? map['image'] as String : null,
      mapImage: map['map_image'] != null ? map['map_image'] as String : null,
    );
  }
}
